packages/@aws-cdk/aws-eks/lib/cluster-resource-handler/cluster.ts

```diff
@@ -22,8 +22,24 @@ export class ClusterResourceHandler extends ResourceHandler {
   constructor(eks: EksClient, event: ResourceEvent) {
     super(eks, event);
 
-    this.newProps = parseProps(this.event.ResourceProperties);
-    this.oldProps = event.RequestType === 'Update' ? parseProps(event.OldResourceProperties) : { };
+    function patchEndpointAccess(props: any) {
+
+      // this is weird but these boolean properties are passed by CFN as a string, and we need them to be booleanic for the SDK.
+      // Otherwise it fails with 'Unexpected Parameter: params.resourcesVpcConfig.endpointPrivateAccess is expected to be a boolean'
+
+      if (typeof(props.resourcesVpcConfig?.endpointPrivateAccess) === 'string') {
+        Object.assign(props.resourcesVpcConfig, { endpointPrivateAccess: props.resourcesVpcConfig.endpointPrivateAccess === 'true' });
```

Comment:
Why do we need to use `assign` here? Can't we just do

```ts
props.resourcesVpcConfig.endpointPrivateAccess = ...
```

Reply:
Right, I got used to working with typed readonly properties. I'll change.